frontend/src/components/ImplementationReportView/EmbedBadges.tsx

```diff
@@ -16,90 +17,180 @@ const EmbedBadges: React.FC<{
 
   const [activeFormat, setActiveFormat] = useState(supportedFormats[1]);
   const [activeBadge, setActiveBadge] = useState(allBadges.Metadata[0]);
-
+  const [show, setShow] = useState(false);
   const badgeEmbed = activeFormat.generateEmbed(activeBadge);
 
   return (
-    <DropdownButton title="Badges" size="sm" align="end" variant="info">
-      <Container className="p-5">
-        <h5 className="pb-1">Badges</h5>
-
-        <Stack className="pb-3" direction="horizontal" gap={5}>
-          <ListGroup variant="flush">
-            {Object.entries(allBadges).map(([category, badges]) => (
-              <ListGroup.Item key={category}>
-                <h6>{category}</h6>
+    <>
+      <Button variant="info" size="sm" onClick={() => setShow(true)}>
+        Badges
+      </Button>
+      <Modal
+        size="xl"
+        fullscreen="lg-down"
+        show={show}
+        onHide={() => setShow(false)}
+      >
+        <Modal.Header closeButton className="border-0 p-0 px-4 px-md-5 pt-5">
+          <Modal.Title className="fs-5">Badges</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Container className="d-none d-sm-block p-2 pb-1 p-lg-5 pt-lg-0">
+            <Stack className="pb-3 gap-2 gap-lg-5" direction="horizontal">
+              <ListGroup variant="flush">
+                {Object.entries(allBadges).map(([category, badges]) => (
+                  <ListGroup.Item key={category}>
+                    <h6>{category}</h6>
+                    <ListGroup variant="flush">
+                      {badges.map((badge) => (
+                        <ListGroup.Item
+                          key={badge.name}
+                          action
+                          // FIXME: wut? badge === activeBadge is false, at
+                          //        least because badge.uri !== activeBadge.uri
+                          //        URI.js has a .equal method
+                          active={badge.name === activeBadge.name}
+                          onClick={() => setActiveBadge(badge)}
+                        >
+                          {badge.name}
+                        </ListGroup.Item>
+                      ))}
+                    </ListGroup>
+                  </ListGroup.Item>
+                ))}
+              </ListGroup>
+
+              <div className="overflow-hidden">
+                <div className="pb-3">
```

Comment:
I haven't stared at this for long but there's obvious duplication here -- I assume for desktop and mobile? Is there not a way you can think of to remove it?

Reply:
Here's a quick prototype for layout without duplication which uses a flex box.

https://github.com/bowtie-json-schema/bowtie/assets/43758514/a06f200a-05ce-4922-9901-482281560ec0

Reply:
If the above layout looks good then I will push it.

Reply:
I don't like the two selectors being next to each other, I put them on opposite sides intentionally, though I definitely could believe further improvements are possible, but putting them on the same side I think is a step backwards.

One selector is something someone will definitely touch (the "which badge do I need one"), and likely touch >1 time (because probably they'll want more than one badge).

One selector *most users* will not even need to touch at all, because we default to Markdown, and 95% of all users will want a Markdown badge.

Putting the two next to each other makes it way easier to click a different thing than you wanted to, and also is just messy feeling to me for a reason I can't quite put a finger on. I wanted to even use a different kind of widget for the two given the above, but couldn't think of a different one.

None of the above has anything to do with mobile though, it's just something I see you changed, not sure why.

On mobile, I think the most logical "small" change is to have things be vertical rather than horizontal.

Reply:
Thanks for explaining me the UX perspective of the UI. I have made changes accordingly. I also tried to implement syntax highlighting for consistency but It is not working `badgeEmbed` template string.


https://github.com/bowtie-json-schema/bowtie/assets/43758514/52abd8ef-cb7e-4099-a340-ebf0287b5417

Reply:
Can you put the padding back on the code block? Having one single line is really hard to interact with, you can't scroll unless you put the cursor exactly on the line, instead of having a big region -- it's not like there's anything else near there, so it seemed very likely that the box should have lots of vertical padding even if the code is just one line.

Reply:
Thanks for the feedback, I have added vertical padding for the code block.

Reply:
Current version of the modal.

![image](https://github.com/bowtie-json-schema/bowtie/assets/43758514/f2c3dad7-a0a8-4c05-a691-cb1d5b457d3a)